UserDataProcessingFindFailureReasonActivity/handler.ts

```diff
@@ -0,0 +1,133 @@
+import { fromNullable } from "fp-ts/lib/Either";
+
+// Activity input
+export const ActivityInput = t.interface({
+  choice: UserDataProcessingChoice,
+  fiscalCode: FiscalCode
+});
+export type ActivityInput = t.TypeOf<typeof ActivityInput>;
+
+// Activity failed because of invalid input
+export const ActivityResultInvalidInputFailure = t.interface({
+  kind: t.literal("INVALID_INPUT_FAILURE"),
+  reason: t.string
+});
+export type ActivityResultInvalidInputFailure = t.TypeOf<
+  typeof ActivityResultInvalidInputFailure
+>;
+
+// Activity failed because of invalid input
+export const ActivityResultNotFoundFailure = t.interface({
+  kind: t.literal("NOT_FOUND_FAILURE")
+});
+export type ActivityResultNotFoundFailure = t.TypeOf<
+  typeof ActivityResultNotFoundFailure
+>;
+
+// Activity failed because of invalid input
+export const ActivityResultUnhandledFailure = t.interface({
+  kind: t.literal("UNHANDLED"),
+  reason: t.string
+});
+export type ActivityResultUnhandledFailure = t.TypeOf<
+  typeof ActivityResultUnhandledFailure
+>;
+
+// Activity failed because of an error on a query
+export const ActivityResultQueryFailure = t.intersection([
+  t.interface({
+    kind: t.literal("QUERY_FAILURE"),
+    reason: t.string
+  }),
+  t.partial({ query: t.string })
+]);
+export type ActivityResultQueryFailure = t.TypeOf<
+  typeof ActivityResultQueryFailure
+>;
+
+export const ActivityResultFailure = t.taggedUnion("kind", [
+  ActivityResultQueryFailure,
+  ActivityResultInvalidInputFailure,
+  ActivityResultNotFoundFailure,
+  ActivityResultUnhandledFailure
+]);
+export type ActivityResultFailure = t.TypeOf<typeof ActivityResultFailure>;
+
+// Activity result
+export const ActivityResultSuccess = t.interface({
+  kind: t.literal("SUCCESS"),
+  value: NonEmptyString
+});
+export type ActivityResultSuccess = t.TypeOf<typeof ActivityResultSuccess>;
+
+export const ActivityResult = t.taggedUnion("kind", [
+  ActivityResultSuccess,
+  ActivityResultFailure
+]);
+
+export type ActivityResult = t.TypeOf<typeof ActivityResult>;
+
+export const getFindFailureReasonActivityHandler = async (
+  context: Context,
+  input: unknown
+): Promise<ActivityResult> => {
+  const client = df.getClient(context);
+
+  return fromEither(ActivityInput.decode(input))
+    .mapLeft(_ => {
+      return ActivityResultFailure.encode({
+        kind: "INVALID_INPUT_FAILURE",
+        reason: "Input not valid for this activity"
+      });
+    })
+    .chain(({ choice, fiscalCode }) => {
+      // create failed orchestrator id based on choice
+      const failedUserDataProcessingOrchestratorId =
+        choice == UserDataProcessingChoiceEnum.DELETE
+          ? makeDeleteOrchestratorId(fiscalCode)
+          : makeDownloadOrchestratorId(fiscalCode);
+
+      return tryCatch(
+        () => {
+          // get the status of the failed orchestrator
+          return client.getStatus(failedUserDataProcessingOrchestratorId);
```

Comment:
I don't understand why we need to read the actual status here. 
Besides of that, when using comment try to tell the _why_, the reason behind the choice you made; the _how_ is already told by code itself ;)

Reply:
Because I found that the orchestrator status contains its output. For old failed user data processing requests this output has the reason of the failure, or at least I think so based on data available to me (some tests done locally). What do you think about? Should I look elsewhere to get the reason?